### PR TITLE
🎉 New Source: KYVE 

### PR DESCRIPTION
**Pull request by cgkournelos-rf**
## What
New source connector for [KYVE](https://www.kyve.network/) Network. Related issue #26222. 

## How
*Developed using (Configuration Based Source) low-code CDK*

## Tests

Acceptance Test Results : 
image

## 🚨 User Impact 🚨
No breaking changes

## Pre-merge Actions
*Expand the relevant checklist and delete the others.*

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests#pull-request-title-convention)
- [x]  [Breaking changes are considered](https://docs.airbyte.com/contributing-to-airbyte/#breaking-changes-to-connectors). If a Breaking Change is being introduced, ensure an Airbyte engineer has created a Breaking Change Plan and you've followed all steps in the [Breaking Changes Checklist](https://docs.airbyte.com/contributing-to-airbyte/#checklist-for-contributors)
- [x]  Connector version has been incremented in the Dockerfile and metadata.yaml according to our [Semantic Versioning for Connectors](https://docs.airbyte.com/contributing-to-airbyte/#semantic-versioning-for-connectors) guidelines
- [x]  Secrets in the connector's spec are annotated with airbyte_secret
- [x]  All documentation files are up to date. (README.md, bootstrap.md, docs.md, etc...)
- [x]  Changelog updated in docs/integrations/<source or destination>/<name>.md with an entry for the new version. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [x]  You, or an Airbyter, have run /test successfully on this PR - or on a non-forked branch
- [x]  You, or an Airbyter, have run /publish successfully on this PR - or on a non-forked branch
- [x]  You've updated the connector's [metadata.yaml](https://docs.airbyte.com/connector-development/connector-metadata-file) file new!

</details>

### CODE REVIEW

**Comment by marcosmarxm** [comment]
@cgkournelos-rf make sure all contributors sign the CLA.

File: airbyte-cdk/python/airbyte_cdk/sources/declarative/declarative_component_schema.yaml [hunk 1]
```diff
@@ -1533,6 +1532,7 @@ definitions:
           - type: string
         interpolation_context:
           - config
+          - response
```
Comment (sherifnada) [inline airbyte-cdk/python/airbyte_cdk/sources/declarative/declarative_component_schema.yaml:1535]: @cgkournelos-rf what is the CDK change being made here? could you split it out into its own PR?

File: airbyte-integrations/connectors/source-kyve/metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+data:
+  connectorSubtype: api
+  connectorType: source
+  # definitionId: decd338e-5647-4c0b-adf4-da0e75f5a750 #TODO
```
Comment (cgkournelos-rf) [inline airbyte-integrations/connectors/source-kyve/metadata.yaml:4]: @sherifnada We are not sure how the definitionId is defined

File: airbyte-cdk/python/airbyte_cdk/sources/declarative/declarative_component_schema.yaml (hunk 1 above)
Comment (cgkournelos-rf) [reply airbyte-cdk/python/airbyte_cdk/sources/declarative/declarative_component_schema.yaml:1535]: This is not something that we changed. It seems that it was brought with the last rebase onto the master. We will fix it!

Comment (cgkournelos-rf) [reply airbyte-cdk/python/airbyte_cdk/sources/declarative/declarative_component_schema.yaml]: Fixed

**Comment by marcosmarxm** [comment]
> @cgkournelos-rf make sure all contributors sign the CLA.

@cgkournelos-rf make sure you check this comment.

File: airbyte-integrations/connectors/source-kyve/metadata.yaml (hunk 2 above)
Comment (cgkournelos-rf) [reply airbyte-integrations/connectors/source-kyve/metadata.yaml]: We run again the latest generate.sh of CDK and we took the given definitionId

**Comment by cgkournelos-rf** [comment]
@marcosmarxm We are ready for the review!!!

One question from our side is how to trigger the GitHub action for updating the checklist?

**Comment by cgkournelos-rf** [comment]
@marcosmarxm This PR is ready for review and from our side, we have completed the checklist.

We are not sure how to trigger the CI test for completing the checklist.

File: airbyte-integrations/connectors/source-kyve/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+RUN apk --no-cache add bash
+
+# copy payload code only
+COPY main.py ./
+COPY source_kyve ./source_kyve
+
+ENV AIRBYTE_ENTRYPOINT "python /airbyte/integration_code/main.py"
+ENTRYPOINT ["python", "/airbyte/integration_code/main.py"]
+
+LABEL io.airbyte.version=0.0.1
```
Comment (sajarin) [inline airbyte-integrations/connectors/source-kyve/Dockerfile:37]: Dockerfile version should match metadata.yaml version (new connectors should start their versioning at 0.1.0)

File: docs/integrations/sources/kyve.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+2. You can specify a specific **`Bundle-Start-ID`** in case you want to narrow the records that will be retrieved from the pool. You can find the valid bundles of in the KYVE app (e.g. [Moonbeam pool bundles](https://app.kyve.network/#/pools/0/bundles)).
+
+## Multiple pools
+You can fetch with one source configuration more than one pool simultaneously. You just need to specify the **`Pool-IDs`** and the **`Bundle-Start-ID`** for the KYVE storage pool you want to archive separated with comma.
+
+## Changelog
+
+| Version | Date | Pull Request | Subject          |
+| :------ | :--- | :----------- | :--------------- |
+| 0.0.1   | 25-05-29  | [26299](https://github.com/airbytehq/airbyte/pull/26299) | Initial release of KYVE source connector|
```
Comment (sajarin) [inline docs/integrations/sources/kyve.md:23]: Changelog version should match metadata.yaml version (new connectors should start their versioning at 0.1.0)

File: airbyte-integrations/connectors/source-kyve/source_kyve/spec.yaml [hunk 5]
```diff
@@ -0,0 +1,43 @@
+documentationUrl: https://docs.airbyte.com/integrations/sources/kyve
+connectionSpecification:
+  $schema: http://json-schema.org/draft-07/schema#
+  title: KYVE Spec
+  type: object
+  required:
+    - pool_ids
+    - start_ids
+  properties:
```
Comment (sajarin) [inline airbyte-integrations/connectors/source-kyve/source_kyve/spec.yaml:9]: Can you add an order property to each of these properties in the spec?

File: airbyte-integrations/connectors/source-kyve/source_kyve/schemas/bitcoin/block.json [hunk 6]
```diff
@@ -0,0 +1,83 @@
+{
+  "$schema" : "http://json-schema.org/draft-04/schema#",
+  "type" : "object",
+  "properties" : {
+    "hash" : {
+      "type" : "string"
+    },
+    "height" : {
```
Comment (sajarin) [inline airbyte-integrations/connectors/source-kyve/source_kyve/schemas/bitcoin/block.json:8]: For each schema, can you add "AdditionalProperties": True?

File: airbyte-integrations/connectors/source-kyve/source_kyve/schemas/bitcoin/block.json [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+    },
+    "versionHex" : {
+      "type" : "string"
+    },
+    "merkleroot" : {
+      "type" : "string"
+    },
+    "time" : {
+      "type" : "integer"
+    },
```
Comment (sajarin) [inline airbyte-integrations/connectors/source-kyve/source_kyve/schemas/bitcoin/block.json:22]: For each of the fields in the schema files, can you modify it so that they are nullable? This will help make the connector less likely to break with schema changes.

**Review by sajarin — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-kyve/Dockerfile (hunk 3 above)
Comment (cgkournelos-rf) [reply airbyte-integrations/connectors/source-kyve/Dockerfile]: Version number fixed

File: docs/integrations/sources/kyve.md (hunk 4 above)
Comment (cgkournelos-rf) [reply docs/integrations/sources/kyve.md]: Resolved

File: airbyte-integrations/connectors/source-kyve/source_kyve/schemas/bitcoin/block.json (hunk 6 above)
Comment (cgkournelos-rf) [reply airbyte-integrations/connectors/source-kyve/source_kyve/schemas/bitcoin/block.json]: Resolved

File: airbyte-integrations/connectors/source-kyve/source_kyve/schemas/bitcoin/block.json (hunk 7 above)
Comment (cgkournelos-rf) [reply airbyte-integrations/connectors/source-kyve/source_kyve/schemas/bitcoin/block.json]: Resolved

File: airbyte-integrations/connectors/source-kyve/source_kyve/spec.yaml (hunk 5 above)
Comment (cgkournelos) [reply airbyte-integrations/connectors/source-kyve/source_kyve/spec.yaml:9]: Done

**Review by sajarin — APPROVED** [review]

**Comment by cgkournelos-rf** [comment]
@sajarin thanks for the feedback. Sure it will be great if you want to test it on your side. However, we are not sure what kind of credentials you require.

**Review by sajarin — CHANGES_REQUESTED** [review]
Just an additional minor comment

**Comment by marcosmarxm** [comment]
Merged at #27373
